parts/k8s/kuberneteswindowssetup.ps1

```diff
@@ -182,67 +182,77 @@ try
     if ($true) {
         Write-Log "Provisioning $global:DockerServiceName... with IP $MasterIP"
 
-        $global:globalTimer = [System.Diagnostics.Stopwatch]::StartNew()
-
-        $configAppInsightsClientTimer = [System.Diagnostics.Stopwatch]::StartNew()
-        # Get app insights binaries and set up app insights client
-        mkdir c:\k\appinsights
-        DownloadFileOverHttp -Url "https://globalcdn.nuget.org/packages/microsoft.applicationinsights.2.11.0.nupkg" -DestinationPath "c:\k\appinsights\microsoft.applicationinsights.2.11.0.zip"
-        Expand-Archive -Path "c:\k\appinsights\microsoft.applicationinsights.2.11.0.zip" -DestinationPath "c:\k\appinsights"
-        $appInsightsDll = "c:\k\appinsights\lib\net46\Microsoft.ApplicationInsights.dll"
-        [Reflection.Assembly]::LoadFile($appInsightsDll)
-        $conf = New-Object "Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration"
-        $conf.DisableTelemetry = -not $global:enableTelemetry

```

Comment:
I believe this line disables the sending of telemetry.  https://docs.microsoft.com/en-us/dotnet/api/microsoft.applicationinsights.extensibility.telemetryconfiguration.disabletelemetry?view=azure-dotnet

If that is the case it makes the resulting code much simpler to reason about as the telemetry is not sent even if the calls are made.  What is the idea for removing the use all together?

Reply:
In some custom clouds, it cannot download the application insight client so it will fail even if it is disabled.

Reply:
We should add the application client to the VHD cache then.  The `DownloadFileOverHttp` checks the cache first https://github.com/Azure/aks-engine/blob/a1870f9a3ec6668515065df1bd41c6d37bff132b/parts/k8s/kuberneteswindowsfunctions.ps1#L27-L30

Reply:
We have a no outbound tests that we run before releases: https://github.com/Azure/aks-engine/blob/master/test/e2e/test_cluster_configs/no_outbound.json#L32 

It might be because telemetry isn't enabled it doesn't catch it.

Reply:
It is in the vhd cache: https://github.com/Azure/aks-engine/blob/f321785bd81b178dcff02736bf3c4799eeaf1c54/vhd/packer/configure-windows-vhd.ps1#L82  

Maybe there is something else causing the outbound traffic?

Reply:
1. Skipping installation and configuration can reduce the provisioning time when telemetry is disabled.
2. Later if we remove hardcoded versions from this script, it also can avoid that mismatched version causes unexpected egress.

Reply:
1.  Do you see a huge provision time improvement? I wouldn't think that the configuration has a significant time difference but I might be wrong.

My biggest trouble with this is the added if statements make the code a bit more difficult to reason and brittle, and if it is disabled we are not seeing this as the egress issue unless I am missing something.  We run [no-out-bound](https://github.com/Azure/aks-engine/blob/master/test/e2e/test_cluster_configs/no_outbound.json) tests as part of the release process which has telemetry disabled and we don't see outbound traffic.

As a way forward it be better if all this logic was in a separate module that hides the if statements?

Reply:
I think that the issue is 
```$global:EnableTelemetry = "{{WrapAsVariable "enableTelemetry" }}";```
=>
```$global:EnableTelemetry = [System.Convert]::ToBoolean("{{WrapAsVariable "enableTelemetry" }}");```

Reply:
I did test that setting $conf.DisableTelemetry will prevent sending telemetry so I think @AbelHu is correct that the issue is the system.convert missing.

If that fixes the issue can we remove all of the added if statements from this PR?